src/util.ts

```diff
@@ -0,0 +1,25 @@
+import { from_hcs0, to_hcs0 }		from '@holo-host/wasm-key-manager';
+import multihash			from 'multihashes';
+
+const Codec = {
+    AgentId: {
+        decode: from_hcs0,
+        encode: to_hcs0,
+    },
+    Signature: {
+        decode: (str) => Buffer.from(str, 'base64'),
+        encode: (buf) => Buffer.from(buf).toString('base64'),
+    },
+    Digest: {
+        decode: (str) => multihash.decode(multihash.fromB58String(str)).digest,
+        encode: (buf) => multihash.toB58String(multihash.encode(Buffer.from(buf), 'sha2-256')),
+    },
+};
+
+const sha256 = async (buf) => {
+    if ( typeof buf === "string" )
+	buf = Buffer.from(buf);
+    return Buffer.from(await crypto.subtle.digest("SHA-256", buf));
```

Comment:
```suggestion
    return Buffer.from(await crypto.subtle.digest("SHA-256", Buffer.from(buf)));
```

And then conditional cast above can be dropped. `Buffer.from` on a `Buffer` is a zero-copy no-op, see docs: https://nodejs.org/api/buffer.html#buffer_class_method_buffer_from_arraybuffer_byteoffset_length

Reply:
This is fixed in commit `5f4c3f0`